ffc/billing/notification_helper.py

```diff
@@ -0,0 +1,149 @@
+import calendar
+import logging
+import textwrap
+from datetime import date
+
+from ffc.billing.dataclasses import (
+    NotificationLevel,
+    ProcessResult,
+    ProcessResultInfo,
+    ProcessResultSummary,
+)
+from ffc.notifications import (
+    NotificationDetails,
+    send_exception,
+    send_info,
+    send_warning,
+)
+
+logger = logging.getLogger(__name__)
+
+NOTIFICATION_FUNCTIONS = {
+    NotificationLevel.SUCCESS: send_info,
+    NotificationLevel.IN_PROGRESS: send_warning,
+    NotificationLevel.ERROR: send_exception,
+}
+
+
+def _build_notification_details(level: NotificationLevel, details: list) -> NotificationDetails:
+    """
+    This function builds a NotificationDetails object depending on the
+    given notification level.
+    """
+
+    if level == NotificationLevel.SUCCESS:
+        return NotificationDetails(
+            ("Authorization", "Journal"),
+            [
+                (
+                    f"{item['authorization_id']}",
+                    f"{item['journal_id'] or '-'}",
+                )
+                for item in details
+            ],
+        )
+    else:
+        return NotificationDetails(
+            ("Authorization", "Journal", "Status", "Message"),
+            [
+                (
+                    f"{item['authorization_id']}",
+                    f"{item['journal_id'] or '-'}",
+                    f"{item['status'].value.upper()}",
+                    "\n\n".join(textwrap.wrap(item.get("message") or "-", width=80)),
+                )
+                for item in details
+            ],
+        )
+
+
+async def _send_notification(
+    level: NotificationLevel, title: str, text: str, results_counter_details: list
+) -> None:
+    """
+    This function sends a notification at the given level.
+    """
+    func = NOTIFICATION_FUNCTIONS[level]
+    await func(
+        title=title,
+        text=text,
+        details=_build_notification_details(level=level, details=results_counter_details),
+    )
+
+
+def count_process_results(
+    results: list[ProcessResultInfo],
+) -> ProcessResultSummary:
+    """
+    This function process the given results list to calculate
+    the number of journals successfully generated and the number of errors.
+    """
+
+    successful_counter = 0
+    error_counter = 0
+    details = []
+
+    for item in (r for r in results if r is not None):
+        successful_counter += item.result == ProcessResult.JOURNAL_GENERATED
+        error_counter += bool(item.message)
+        details.append(
+            {
+                "authorization_id": item.authorization_id,
+                "journal_id": item.journal_id,
+                "status": item.result,
+                "message": item.message,
+            }
+        )
+    return ProcessResultSummary(
```

Comment:
I think there is no need here for a dataclass. 
We should just split the list into two lists one that contains the positive ProcessResultInfo one with the negative like:


```python
succeeded, failed = [], []

for ri in results:
    if ri.result == ProcessResult.JOURNAL_GENERATED:
        succeeded.append(ri)
    else:
        failed.append(ri)
    return succeeded, failed
 ```